rust/agents/relayer/src/msg/serial_submitter.rs

```diff
@@ -196,38 +196,70 @@ impl SerialSubmitter {
             .run_queue_length_gauge
             .set(self.run_queue.len() as i64);
 
-        // Deliver the highest-priority message on the run queue.
-        if let Some(mut msg) = self.run_queue.pop() {
-            info!(msg=?msg, "ready to deliver message");
-            match self.deliver_message(&msg).await {
-                Ok(()) => {
-                    info!(msg=?msg, "message delivered");
-                }
-                Err(e) => {
-                    info!(msg=?msg, "message delivery failed: {}", e);
-                    msg.num_retries += 1;
-                    self.run_queue.push(msg);
-                }
+        // Pick the next message to try delivering.
+        let mut msg = match self.run_queue.pop() {
+            Some(m) => m,
+            None => return Ok(()),
+        };
+
+        // If the message has already been delivered according to message_status call on
+        // inbox, e.g. due to another relayer having already delivered, then mark it as
+        // already-delivered, and move on to the next tick.
+        // TODO(webbhorn): Make this robust to re-orgs on inbox.
+        if let MessageStatus::Processed = self
+            .inbox_contracts
+            .inbox
+            .message_status(msg.committed_message.to_leaf())
```

Comment:
Shouldn't we check the status in the db before making this call?

Reply:
We already check the DB status in processor.rs (see MessageProcessor::tick() near log message "Skipping since message_index already in DB"). If our local DB claims that the message has already been delivered, then the submitter will never even see the message as a candidate.